src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,1473 @@
+                buildChangeTargetUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).contains(ChangeTarget.of(null));
+    }
+
+    @Test
+    void buildChangeTargetUpdateAction_WithJustOldTargetHasNullValue_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getTarget()).thenReturn(null);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final LineItemsTarget newTarget = LineItemsTarget.ofAll();
+        when(newCartDiscountDraft.getTarget()).thenReturn(newTarget);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+                buildChangeTargetUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).contains(ChangeTarget.of(newTarget));
+    }
+
+    @Test
+    void buildChangeTargetUpdateAction_WithBothTargetHaveNullValues_ShouldNotBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getTarget()).thenReturn(null);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getTarget()).thenReturn(null);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+                buildChangeTargetUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).isNotPresent();
+    }
+
+    @Test
+    void buildChangeIsActiveUpdateAction_WithDifferentValues_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.isActive()).thenReturn(false);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.isActive()).thenReturn(true);
+
+        final Optional<UpdateAction<CartDiscount>> changeIsActiveUpdateAction =
+                buildChangeIsActiveUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeIsActiveUpdateAction).contains(ChangeIsActive.of(true));
+    }
+
+    @Test
+    void buildChangeIsActiveUpdateAction_WithSameValues_ShouldNotBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.isActive()).thenReturn(false);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.isActive()).thenReturn(false);
+
+        final Optional<UpdateAction<CartDiscount>> changeIsActiveUpdateAction =
+                buildChangeIsActiveUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeIsActiveUpdateAction).isNotPresent();
+    }
+
+    @Test
+    void buildChangeIsActiveUpdateAction_WithJustNewIsActiveHasNullValue_ShouldBuildUpdateAction() {
```

Comment:
`WithOnlyNullNewIsActive`

Reply:
what about the case if new is `null` and old is `true`?

Reply:
Actually, if it's null it will not create update action because the default value for the isActive is true. But I realized, I removed the test for it, I have added it back : 92d640a
https://github.com/commercetools/commercetools-sync-java/blob/860c750f8c6afa2b46ef472eda3c0528961d94fa/src/main/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtils.java#L183-L197

Reply:
😄  but why did you remove the old test when the old is `false`?  I understand what would happen, but we should have tests for both cases.

Reply:
I will add these tests.

Reply:
5bd6ba4